src/Umbraco.Web/PublishedCache/NuCache/ITransactableDictionaryFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Umbraco.Web.PublishedCache.NuCache
+{
+    public interface ITransactableDictionaryFactory
+    {
+        ITransactableDictionary<int, ContentNodeKit> Get(ContentCacheEntityType entityType);
+
+        /// <summary>
+        /// Ensures that the ITransactableDictionaryFactory has the proper environment to run.
+        /// </summary>
+        /// <param name="errors">The errors, if any.</param>
+        /// <returns>A value indicating whether the ITransactableDictionaryFactory has the proper environment to run.</returns>
+        bool EnsureEnvironment(out IEnumerable<string> errors);
+
+        /// <summary>
+        /// Whether the dictionary has been populated
+        /// </summary>
+        /// <returns></returns>
+        bool IsPopulated(ContentCacheEntityType entityType);
```

Comment:
What is the reason for having IsPopulated on both the ITransactableDictionaryFactory and the ITransactableDictionary? I can only see the factory usage like `if ((_localContentDb != null && _localContentDb.IsPopulated()) || _transactableDictionaryFactory.IsPopulated(ContentCacheEntityType.Document))` but I'm unsure why?